staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go

```diff
@@ -94,7 +109,28 @@ type Pool struct {
 	// by the controller.
 	Generation int64
 
-	// Device names must be unique inside the pool.
+	// Slices is a list of all ResourceSlices that the driver
+	// wants to publish for this pool. The driver must ensure
+	// that each resulting slice is valid. See the API
+	// definition for details, in particular the limit on
+	// the number of devices.
+	//
+	// If slices are not valid, then the controller will
+	// log errors produced by the apiserver.
+	//
+	// Drivers should publish at least one slice for each
+	// pool that they normally manage, even if that slice
+	// is empty. "Empty pool" is different from "no pool"
+	// because it shows that the driver is up-and-running
+	// and simply doesn't have any devices.
```

Comment:
@klueska: you wanted to check whether this "should publish" is sufficient or whether we should enforce this by creating an empty slice when there is none.

Reply:
I have a feeling driver authors will get this wrong as it's not very intuitive to *have* to create an empty slice if you aren't actually publishing any devices.

Reply:
That mirrors my own thinking, so let's leave it as-is for now.